segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/TableService.java

```diff
@@ -48,9 +48,9 @@ public TableService(SegmentContainerRegistry segmentContainerRegistry, SegmentTo
     //region TableStore Implementation
 
     @Override
-    public CompletableFuture<Void> createSegment(String segmentName, Duration timeout) {
+    public CompletableFuture<Void> createSegment(String segmentName, boolean sorted, Duration timeout) {
```

Comment:
We could very well do that, but sorting doesn't come for free. There will be additional space requirement (keep all keys in sorted order) and some computational overhead to maintain that sorted index. Not all Table Segments may want that. If all they want is a Hash-Table-like structure with no order between keys, then regular Table Segments will do the job without the overhead of sorted index.

Additionally, there is no upgrade path for Hash Table Segments to Sorted Table Segments. IF we converted all HTS into STS, then we'd have to auto-upgrade existing Segments. That would be another challenge by its own.

Reply:
I guess we don't have a concrete answer to this, but I'm wondering about the recommendation we would give about the use of STS, like what's a reasonable number of sorted tables or what is a reasonable amount of traffic to guide users. 

The migration is indeed a concern.

Reply:
For now, I think we should keep all metadata (Segment Store and Controller) in hash tables - there is no need for sorted keys there. 

For KVTs, we will use STSs because we want to use iterators. In the (near) future, we can add an option at the creation of KVTs that says "I don't need iterators" which should result in the use of Hash Table Segments as backing store (the API is identical between STS and HTS). 

* I say "near future" because KVTs are going out as **Beta**, which gives us certain freedoms in altering the API without need for backwards compatibility while having this tag.